memtable/write_buffer_manager.cc

```diff
@@ -84,26 +117,25 @@ void WriteBufferManager::ReserveMemWithCache(size_t mem) {
 }
 
 void WriteBufferManager::ScheduleFreeMem(size_t mem) {
+  std::unique_lock<std::mutex> lock(wbm_mutex_);
```

Comment:
IIRC, we don't use lock for `memory_active_` and other similar variables to reduce the locking/releasing the mutex which can be expensive and these variables are already atomic so they don't really need the mutex.

Reply:
Since we are reading the `buffer_size_`'s value with order `std::memory_order_relaxed`, if without locking, do we concern about not getting the latest modified value by other thread to `buffer_size_` through `SetBufferSize()` because of  `std::memory_order_relaxed`? 

This is another reason in addition to "making related operations atomic" for simplicity as mentioned in https://github.com/facebook/rocksdb/pull/11253/files#r1151045690.

This reminds me of we should sync up on whether we care about these two things as the answer to these questions can impact this PR e.g, remove some lock added